lms/djangoapps/discussion/views.py

```diff
@@ -739,11 +739,13 @@ def render_to_fragment(
             return fragment
         except cc.utils.CommentClientMaintenanceError:
```

Comment:
@awaisdar001 
The fragment passed at #R734 is using already using an html param that is already rendered to string. The reason for AssertionError was due to passing render_to_response() object as a parameter.
CommentClientMaintenanceError is raised from https://github.com/edx/edx-platform/blob/068b4a8c44e3c0a3d17ef564651d20427cbd0b80/lms/lib/comment_client/utils.py#L83

Reply:
I have tested this on sandbox and it seems like edx is already showing a generic message against such errors. 

<img width="1440" alt="screen shot 2019-02-04 at 7 57 45 pm" src="https://user-images.githubusercontent.com/4076623/52339613-6f42f780-2a2f-11e9-8747-4641fdb3ec1e.png">
As you can see in the screen shot that console shows 500 error and logs show the AssertionError
<img width="903" alt="screenshot 2019-02-04 at 8 00 55 pm" src="https://user-images.githubusercontent.com/4076623/52339797-e082aa80-2a2f-11e9-84d9-858e88b0616b.png">
After fixing this error discussion board will show the following message in case of error:
<img width="778" alt="screenshot 2019-02-04 at 7 49 21 pm" src="https://user-images.githubusercontent.com/4076623/52339877-23448280-2a30-11e9-9923-f26e1d4d3db7.png">

So i think there is no need to explicitly handle this AssertionError for open edx as default 500 page is rendered in case of any error.
what is your opinion about this?

Reply:
@ihtram Is the following screenshot is from open edx sandbox? if it is, it looks like there is 500 error on the server and we would like to avoid that. 

![](https://user-images.githubusercontent.com/4076623/52339613-6f42f780-2a2f-11e9-8747-4641fdb3ec1e.png)


So I think this PR makes sense. Let me know what do you think.

Reply:
@awaisdar001  yes that make sense but if we merge this change it will not show the default message. It will show the error message like the one in attached screen shot

<img width="778" alt="screenshot 2019-02-04 at 7 49 21 pm" src="https://user-images.githubusercontent.com/4076623/52477565-253f4a80-2bc4-11e9-88a2-e2ccd8dc9cd4.png">

If this looks good then we can go ahead and merge this PR.

Reply:
let's go for the warning message "The discussions are currently undergoing maintenance. .... "